app/Http/Controllers/RoomAPIController.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Room;
+use Illuminate\Http\Request;
+
+class RoomAPIController extends Controller
+{
+   public function find(int $id): \Illuminate\Http\JsonResponse
+   {
+       $room = Room::find($id);
+       if($room){
```

Comment:
You should instead check to see if there is not a room here, and then return your 404 response.

It's always good practice to 'fail early', meaning you should check for failures (like the room not existing) as soon as possible.

This way, you can have something like

```
if (!$room) {
   // return the 404
}

// Return the success response
```

Eliminating the need for the else which does help to make the code simpler